state-chain/pallets/cf-pools/src/tests.rs

```diff
@@ -1402,3 +1420,156 @@ fn test_sweeping_when_updating_range_order() {
 		assert_eq!(HistoricalEarnedFees::<Test>::get(BOB, STABLE_ASSET), EXPECTED_FEES);
 	});
 }
+
+#[test]
+fn test_limit_order_expire_at() {
+	const ASSET: Asset = Asset::Flip;
+	const EXPIRE_AT: u64 = 10;
+	const AMOUNT: AssetAmount = 10_000;
+	const ORDER_ID: u64 = 1;
+
+	new_test_ext()
+		.execute_with(|| {
+			assert_ok!(LiquidityPools::new_pool(
+				RuntimeOrigin::root(),
+				ASSET,
+				STABLE_ASSET,
+				0,
+				price_at_tick(0).unwrap(),
+			));
+
+			MockBalance::credit_account(&ALICE, ASSET, AMOUNT);
+
+			// Make sure that a expiry block that is in the past is not accepted
+			assert_noop!(
+				LiquidityPools::set_limit_order(
+					RuntimeOrigin::signed(ALICE),
+					ASSET,
+					STABLE_ASSET,
+					Side::Sell,
+					ORDER_ID,
+					Some(5),
+					AMOUNT,
+					Some(0), // Block 0 will always be in the past
+				),
+				Error::<Test>::LimitOrderUpdateExpired
+			);
+
+			// Set a limit order with an expiry block in the future
+			assert_ok!(LiquidityPools::set_limit_order(
+				RuntimeOrigin::signed(ALICE),
+				ASSET,
+				STABLE_ASSET,
+				Side::Sell,
+				ORDER_ID,
+				Some(5),
+				AMOUNT,
+				Some(EXPIRE_AT),
+			));
+
+			// Check that the event for scheduling the expire order is emitted
+			assert_events_match!(
+				Test,
+				RuntimeEvent::LiquidityPools(Event::LimitOrderSetOrUpdateScheduled {
+					lp: ALICE,
+					order_id: ORDER_ID,
+					dispatch_at: EXPIRE_AT,
+				}) => ()
+			);
+
+			// The order should be present in the pool
+			assert_eq!(
+				LiquidityPools::pool_orders(ASSET, STABLE_ASSET, Some(ALICE), false)
+					.unwrap()
+					.limit_orders
+					.asks
+					.len(),
+				1
+			);
+		})
+		.then_execute_at_block(EXPIRE_AT, |_| {
```

Comment:
Probably better to use `then_process_blocks_until_block` since it is closer to what happens in practice (i.e. we don't skip blocks).